tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -37,13 +37,39 @@
 #include "workload_processor.h"
 
 namespace tcf {
-
     WorkOrderProcessor::~WorkOrderProcessor() {
         // Sanitize class members storing secrets
         worker_encryption_key.clear();
         session_key.clear();
     }
 
+    void WorkOrderProcessor::PopulateExtInfoData(
+        std::string ext_wo_data,
+        ExtWorkOrderInfoKME* ext_wo_info_kme) {
```

Comment:
Please add a function `PopulateExtInfoData` in file `workload_processor.h`. WorkOrderProcesor will call `PopulateExtInfoData()` before calling `ProcessWorkOrder()` by passing all required arguments 

`WorkLoadProcessor::PopulateExtInfoData(std::string ext_wo_data, std::string client_session_key, std::string requester_nonce, std::vector<tcf::WorkOrderData> client_in_wo_keys, std::vector<tcf::WorkOrderData> client_out_wo_keys)`

Reply:
Add a default implementation for `PopulateExtInfoData()`  in `WorkloadProcessor` class. That way only `KMEWorkloadProcessor` class needs to override this function. Populate `ExtWorkOrderInfoKME` structure inside `PopulateExtInfoData()` in `kme_workload_plug-in.cpp`

Reply:
This will avoid the dependency of including `ext_work_order_info_kme.h` in `WorkOrderProcessor`

Reply:
The above mentioned way of populating ExtWorkOrderInfoKME members was to avoid exposing client keys to workload. If we create PopulateExtInfoData API in workloadprocessor and KME workload overrides it, then KME workload can access the keys. Hence, we initialzed member variables inside WorkOrderProcessor itself.

Reply:
I can see below line -> `ext_wo_info_kme->SetWorkOrderSymmetricKey(this->session_key)`. So  issession key is exposed to KME workload ?. KME Enclave can give enc_sym_key to KME workload. Please replace `client_session_key` by `enc_client_session_key`

Reply:
Please create KME work order processor (inherit from work_order_processor.cpp) and implement KME specific work order processing in that class. That will help to organize the work order processing of of singleton/WPE/KME better

Reply:
I have created separate WorkOrderProcessorKME class which inherits from WorkOrderProcessor.
I have implemented KME specific functionalities in this class.